backend/app/routes/game_routes.py

```
@@ -75,15 +96,23 @@ def add_weak_point():
+    if not user_id:
+        return jsonify({'error': '認証が必要です'}), 401
+    
     data = request.get_json()
+    if not all(k in data for k in ['level', 'score']):
+        return jsonify({'error': '必須フィールドが不足しています'}), 400
+    
     progress = Progress(
-        user_id=data['user_id'],
+        user_id=user_id,
         level=data['level'],
         score=data['score']
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)